src/epsilonServer.cpp

```diff
@@ -15,8 +16,12 @@ EpsilonServer::EpsilonServer()
     for(unsigned int n=0; n<factionInfo.size(); n++)
         factionInfo[n]->reset();
 
-    state_logger = new GameStateLogger();
-    state_logger->start();
+    if (PreferencesManager::get("game_logs", "1").toInt())
+    {
+        state_logger = new GameStateLogger();
+        state_logger->start();
+    } else if (state_logger)
+        state_logger->destroy();
```

Comment:
The last part about destroying the logger can never trigger. As this is during construction of the server object. The server holds the state_logger when it is created and will destroy the logger as well when the server gets destroyed.

As this is "at creation time", there is never a logger in this member.